add documentation for configuring SASL with the helm chart (#706)
redpanda-data/documentation: docs/security/sasl-kubernetes.mdx

---
title: Configuring Redpanda SASL on Kubernetes
---

<head>
    <meta name="title" content="Configuring Redpanda SASL on Kubernetes | Redpanda Docs"/>
    <meta name="description" content="Configuring Redpanda SASL on Kubernetes."/>
</head>

> Redpanda supports the SASL/SCRAM authentication method for the Kafka API. The Redpanda Helm chart lets you configure SASL authentication.

Author: Let's make this match the other one.

```suggestion
The Simple Authentication and Security Layer (SASL) is a method for adding authentication support to connection-based protocols. Alone, SASL only provides authentication between the server and the client. To encrypt communication between the server and the client, combine SASL authentication with TLS encryption.
```

:::note

> This page uses the recommended Helm chart for configuring security. For information about using the older Redpanda Operator, see [Redpanda Operator](../../reference/redpanda-operator).

Reviewer: ```suggestion
This page uses the recommended Helm chart for configuring security. For information about using previous versions of Redpanda Operator, see [Redpanda Operator](../../reference/redpanda-operator).
```

Reviewer 2: resolved with new notes above

> This page uses the recommended Helm chart for configuring security. For information about using the older Redpanda Operator, see [Redpanda Operator](../../reference/redpanda-operator).

Reviewer: Thought we were removing "older" throughout?
For information about using Redpanda Operator...

Reviewer 2: Ah, thanks. I'd updated the pages in reference, but not these ones in security. Now the notes for the TLS and SASL pages in the security section say (with correct word/link for TSL/SASL): 

This page uses the recommended `redpanda` Helm chart for configuring TLS. For information about using the `redpanda-operator` Helm chart (supported for backward compatibility), see [Redpanda Operator](../../reference/redpanda-operator).

:::

## Enable SASL

Create a YAML file containing the values to override from the defaults:

`sasl_enable.yaml`

```yaml
auth:
  sasl:
    enabled: true
    users:
      - name: admin
        password: changeme
```

:::note
In this YAML document, `users` is a list of superusers.
:::

> During install or upgrade, enable SASL configuration:

Reviewer: What is redpanda is already installed? Or has already been upgraded? Then what should the user do?

Author: It's common Helm parlance to refer to any changes to the initial installation as an upgrade.

```bash
helm upgrade --install redpanda redpanda/redpanda -n redpanda --create-namespace \
  --values sasl_enable.yaml
```

## Create users

Create users (not superusers) and set passwords for the new users. By default, these users don't have any permissions on the cluster.

:::tip

> As a security best practice, you don't want to use the superuser to run commands on the cluster. You can use these additional users to interact with the cluster.

Reviewer: Maybe it's just me, but I don't understand this guidance. Who are "these additional users"?
Should it read: 
As a security best practice, superusers should not run commands on the cluster. Instead, have these additional or new users interact with the cluster.

Author: I read that as the users specified in the intro: "Create users (not superusers)"...

Reviewer 2: replace with Joyce's suggestion

:::

> For each user that you want to create, run:

Reviewer: ```suggestion
For each user that you create, run:
```

Author: ```suggestion
To create the user `myuser` with a password `changethispassword`, run:
```

I think we can count on folks being smart enough to recognize that the command structure is the same for each of multiple users.

```bash
kubectl exec -n redpanda -c redpanda redpanda-0 -- \
  rpk acl user create myuser -p changethispassword
```

## Grant permissions

> The superuser can grant permissions to additional users through access control lists ([ACLs](../authorization/#ACLs)). For details on how ACLs function in Redpanda, see the [rpk acl](../../reference/rpk/rpk-acl/) documentation.

Reviewer: ```suggestion
The superuser can grant permissions to additional users through access control lists ([ACLs](../authorization/#ACLs)). For details on how ACLs function in Redpanda, see [rpk acl](../../reference/rpk/rpk-acl/).
```

1. Use the superuser to grant `create` and `describe` permissions to another user for the cluster. Edit the `rpk acl create` command to grant permissions to specific users or groups:

```bash
kubectl exec -n redpanda -c redpanda redpanda-0 -- \
  rpk acl create --allow-principal User:myuser \
  --operation create,describe \
  --cluster \
  --user admin --password changeme --sasl-mechanism SCRAM-SHA-256

> ```

Reviewer: I believe this code example should be indented to align with the step.
Please confirm @Deflaimun

Reviewer: See how the code box extends out to the left of the step?
![image](https://user-images.githubusercontent.com/102751339/196547594-69df998a-ac7f-4516-91ba-308942ff5ba8.png)

Reviewer 2: I could fix this formatting.

2. Optionally, you can use the superuser to grant permissions to a new user for a topic. The following command grants `describe` privileges to a topic:

```bash
kubectl exec -n redpanda -c redpanda redpanda-0 -- \
  rpk acl create --allow-principal User:myuser \
  --operation describe \
  --topic myfirsttopic \
  --user admin --password changeme --sasl-mechanism SCRAM-SHA-256
```

:::note
If a user has `describe` privileges on a cluster, they don't automatically have `describe` privileges on topics.
:::

> ## Use `rpk` to interact with Redpanda

Author: You can't use code blocks in titles.

```suggestion
## Use rpk to interact with Redpanda
```

I mean you _can_, but they won't be formatted as code blocks.

> If a user has `describe` privileges on a cluster, they don't automatically have `describe` privileges on topics.

Reviewer: ```suggestion
If a user has `describe` privileges on a cluster, it does mean that user automatically has `describe` privileges on topics.
```

Author: ```suggestion
If a user has `describe` privileges on a cluster, it does not mean that user is automatically granted `describe` privileges on topics.
```

> Now you can connect to Redpanda with the additional (non-superuser) user and start working with the cluster.

Reviewer: Hmmm...customers don't always come to a section in a linear fashion (ie. Sequentially reading one topic after another). How about:
"To connect to Redpanda with the additional (non-superuser) user and start working with the cluster:"

Reviewer 2: replaced with Joyce's suggestion

  --user admin --password changeme --sasl-mechanism SCRAM-SHA-256
```

> 2. Optionally, you can use the superuser to grant permissions to a new user for a topic. The following command grants `describe` privileges to a topic:

Author: You just created a user in the previous step. "the" is correct as it's referring to that user. "a" is too vague (why would you be setting permissions for some other new user that you didn't just create?)

Reviewer 2: fixed
